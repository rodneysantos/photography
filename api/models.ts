export interface SearchResponse {
	url: string;
}

export interface Cloudinary {
	total_count: number;
	time: number;
	resources: Resource[];
}

export interface Resource {
	asset_id: string;
	public_id: string;
	folder: string;
	filename: string;
	format: string;
	version: number;
	resource_type: string;
	type: string;
	created_at: string;
	uploaded_at: string;
	bytes: number;
	backup_bytes: number;
	width: number;
	height: number;
	aspect_ratio: number;
	pixels: number;
	url: string;
	secure_url: string;
	status: string;
	access_mode: string;
	access_control: string | null;
	etag: string;
	created_by: Author;
	uploaded_by: Author;
	last_updated: Timestamp;
}

export interface Author {
	access_key: string;
	custom_id: string;
	external_id: string;
}

export interface Timestamp {
	tags_updated_at: string;
	updated_at: string;
}
