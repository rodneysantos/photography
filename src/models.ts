export interface Response {
	data: Photo[];
	error: unknown | null;
}

export interface Photo {
	url: string;
}
