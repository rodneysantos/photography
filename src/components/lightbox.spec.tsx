import { cleanup, render } from "@solidjs/testing-library";
import { afterEach, describe, expect, it, vi } from "vitest";
import { Photo } from "../models";
import { Lightbox } from "./lightbox";

describe("lightbox", () => {
	const onCloseMock = vi.fn();
	const photo: Photo = { url: "photo" };

	afterEach(cleanup);

	afterEach(() => {
		vi.resetAllMocks();
	});

	it("renders correctly", () => {
		// assemble
		const { asFragment } = render(() => (
			<Lightbox photo={photo} onClose={onCloseMock} />
		));

		// assert
		expect(asFragment()).toMatchSnapshot();
	});

	it("close button calls onClose", () => {
		// assemble
		const { getByTestId } = render(() => (
			<Lightbox photo={photo} onClose={onCloseMock} />
		));
		const closeButton = getByTestId("lightbox-close");

		// act
		closeButton.click();

		// assert
		expect(onCloseMock).toHaveBeenCalled();
	});

	it("does not render when photo is undefined", () => {
		// assemble
		const { queryByTestId } = render(() => (
			<Lightbox photo={undefined} onClose={onCloseMock} />
		));

		// assert
		expect(queryByTestId("lightbox")).toBeNull();
	});
});
