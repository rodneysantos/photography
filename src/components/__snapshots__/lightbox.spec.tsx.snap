// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lightbox > renders correctly 1`] = `"<div data-testid=\\"lightbox\\" class=\\"fixed top-0 left-0 z-50 flex items-center justify-center w-full h-full bg-black bg-opacity-60\\"><div class=\\"relative bg-white\\"><div class=\\"absolute top-0 right-0 z-50 p-4\\"><button data-testid=\\"lightbox-close\\" class=\\"p-2 transition-colors rounded-full\\"><svg class=\\"w-6 h-6 text-white\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\"><path stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M6 18L18 6M6 6l12 12\\"></path></svg></button></div><img data-testid=\\"lightbox-photo\\" class=\\"h-full p-2 bg-white min-md:h-screen\\" alt=\\"\\" src=\\"photo\\"></div></div>"`;
